core/platform/transactions/gae_transaction_services.py

```diff
@@ -20,15 +20,39 @@
 
 
 def run_in_transaction(fn, *args, **kwargs):
-    """Run a function in a transaction."""
+    """Run a function in a transaction.
+    Either all of the operations in the transaction are applied,
+    or none of them are applied.
+    If an exception is raised, the transaction is likely not safe to
```

Comment:
E.g. I think this one is a good observation and belongs in its own paragraph, so add one more blank line above this one.